src/contracts/pods/EigenPod.sol

```diff
@@ -244,36 +335,59 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
             "EigenPod.verifyAndProcessWithdrawals: inputs must be same length"
         );
 
-        // Verify passed-in beaconStateRoot against oracle-provided block root:
-        BeaconChainProofs.verifyStateRootAgainstLatestBlockRoot({
-            latestBlockRoot: eigenPodManager.getBlockRootAtTimestamp(oracleTimestamp),
-            beaconStateRoot: stateRootProof.beaconStateRoot,
-            stateRootProof: stateRootProof.proof
-        });
+        // Check for a relevant partial withdrawal checkpoint using the oracle timestamp
+        // For full withdrawals, we update a checkpoint if our withdrawal proof is older
+        // than the checkpoint
+        bool updateCheckpoint = currentCheckpointTimestamp >= oracleTimestamp;
+        Checkpoint memory checkpoint = currentCheckpoint;
+
+        // Get the cached beacon state root for this timestamp, or validate `stateRootProof`
+        bytes32 beaconStateRoot = _getOrValidateStateRoot(oracleTimestamp, stateRootProof);
 
         VerifiedWithdrawal memory withdrawalSummary;
         for (uint256 i = 0; i < withdrawalFields.length; i++) {
+            bytes32 pubkeyHash = validatorFields[i].getPubkeyHash();
+
             VerifiedWithdrawal memory verifiedWithdrawal = _verifyAndProcessWithdrawal(
-                stateRootProof.beaconStateRoot,
+                beaconStateRoot,
+                pubkeyHash,
                 withdrawalProofs[i],
                 validatorFieldsProofs[i],
                 validatorFields[i],
                 withdrawalFields[i]
             );
 
-            withdrawalSummary.amountToSendGwei += verifiedWithdrawal.amountToSendGwei;
+            withdrawalSummary.amountToQueueGwei += verifiedWithdrawal.amountToQueueGwei;
             withdrawalSummary.sharesDeltaGwei += verifiedWithdrawal.sharesDeltaGwei;
-        }
 
-        // If any withdrawals are eligible for immediate redemption, send to the pod owner via
-        // DelayedWithdrawalRouter
-        if (withdrawalSummary.amountToSendGwei != 0) {
-            _sendETH_AsDelayedWithdrawal(podOwner, withdrawalSummary.amountToSendGwei * GWEI_TO_WEI);
+            // If we created a checkpoint after this validator was exited from the beacon chain
+            // and we haven't seen a proof for this validator, mark them as proven and decrement
+            // their `amountToQueueGwei` from the checkpoint's eligible balance
+            if (updateCheckpoint && !validatorProven[oracleTimestamp][pubkeyHash]) {
+                validatorProven[oracleTimestamp][pubkeyHash] = true;
+                checkpoint.proofsRemaining--;
+                checkpoint.eligibleBalance -= verifiedWithdrawal.amountToQueueGwei;
```

Comment:
hmm i think this might open up the

1. repoint val w 32 eth
2. fully withdraw it from beaconchain for 32 eth
3. deposit to it again for 1 eth
4. fully withdraw again from beacon for 1 eth
5. start checkpoint
6. prove 1 eth full withdrawal

now u get 32 eth partial withdrawal for free unstaked. whereas before you would eventually have to restake the 32 eth again.

Reply:
I'm a little confused about the order of operations - at what point are withdrawal credentials proven? Because you need to do that to prove a withdrawal.

Reply:
step 1

Reply:
oh you're talking about the "current issue" i found with eigenpods?

checkpointing doesn't "open up" the issue again. it's always been there and this doesn't change that

Reply:
But it's sort of amplified here. In the current system you must restake the 32 eth again, whereas you would be able to withdraw it in this scenario

Reply:
> you must restake the 32 eth again

by "restake" you mean that you need to provide another withdrawal proof, causing the shares to be re-awarded (and only then withdrawing)

i'm not convinced that's much better. i think the root problem is the same for both, and that's how both of these interact with how slashing is supposed to function with the withdrawal queue

If you can suddenly change the reported backing amount, you can reduce your slashable amount on a whim.

Reply:
Whether you need to "reenter the system" or the eth just leaves the pod, IMO not the crux of the system. Both would need to be accounted for when we build slashing.